diff -Naur coreutils-9.4/lib/posixtm.c coreutils-9.4_patched/lib/posixtm.c
--- coreutils-9.4/lib/posixtm.c	2023-01-01 08:39:11.000000000 -0500
+++ coreutils-9.4_patched/lib/posixtm.c	2024-03-11 14:07:36.479338489 -0400
@@ -191,7 +191,7 @@
              | (tm0.tm_min ^ tm1.tm_min)
              | (tm0.tm_sec ^ tm1.tm_sec)))
         {
-          if (ckd_add (&t, t, leapsec))
+          if (ckd_add (&t, t, +leapsec))
             return false;
           *p = t;
           return true;
