diff --git a/src/main.c b/src/main.c
index 43586c4..df4c114 100644
--- a/src/main.c
+++ b/src/main.c
@@ -30,6 +30,14 @@ extern void jv_tsd_dtoa_ctx_init();
 #define USE_ISATTY
 #endif
 
+#ifdef CUSTOM_VERSION
+#define VERSION_SUFFIX "-" CUSTOM_VERSION
+#else
+#define VERSION_SUFFIX ""
+#endif
+
+#define JQ_CUSTOM_VERSION JQ_VERSION VERSION_SUFFIX
+
 #include "compile.h"
 #include "jv.h"
 #include "jq.h"
@@ -65,7 +73,7 @@ static void usage(int code, int keep_it_short) {
     "the jq(1) manpage (\"man jq\") and/or https://jqlang.github.io/jq/.\n\n"
     "Example:\n\n\t$ echo '{\"foo\": 0}' | jq .\n"
     "\t{\n\t  \"foo\": 0\n\t}\n\n",
-    JQ_VERSION, progname, progname, progname);
+    JQ_CUSTOM_VERSION, progname, progname, progname);
   if (keep_it_short) {
     fprintf(f,
       "For listing the command options, use %s --help.\n",
@@ -575,7 +583,7 @@ int main(int argc, char* argv[]) {
         if (!short_opts) continue;
       }
       if (isoption(argv[i], 'V', "version", &short_opts)) {
-        printf("jq-%s\n", JQ_VERSION);
+        printf("jq-%s\n", JQ_CUSTOM_VERSION);
         ret = JQ_OK;
         goto out;
       }
@@ -650,10 +658,10 @@ int main(int argc, char* argv[]) {
   jq_set_attr(jq, jv_string("JQ_ORIGIN"), jv_string(dirname(origin)));
   free(origin);
 
-  if (strchr(JQ_VERSION, '-') == NULL)
-    jq_set_attr(jq, jv_string("VERSION_DIR"), jv_string(JQ_VERSION));
+  if (strchr(JQ_CUSTOM_VERSION, '-') == NULL)
+    jq_set_attr(jq, jv_string("VERSION_DIR"), jv_string(JQ_CUSTOM_VERSION));
   else
-    jq_set_attr(jq, jv_string("VERSION_DIR"), jv_string_fmt("%.*s-master", (int)(strchr(JQ_VERSION, '-') - JQ_VERSION), JQ_VERSION));
+    jq_set_attr(jq, jv_string("VERSION_DIR"), jv_string_fmt("%.*s-master", (int)(strchr(JQ_CUSTOM_VERSION, '-') - JQ_CUSTOM_VERSION), JQ_CUSTOM_VERSION));
 
 #ifdef USE_ISATTY
   if (!program && (!isatty(STDOUT_FILENO) || !isatty(STDIN_FILENO)))
