diff --git a/Lib/platform.py b/Lib/platform.py
index 7bb2220..15ceace 100755
--- a/Lib/platform.py
+++ b/Lib/platform.py
@@ -477,7 +477,7 @@ def _mac_ver_xml():
     return release, versioninfo, machine
 
 
-def mac_ver(release='', versioninfo=('', '', ''), machine=''):
+def mac_ver(release='14.4.1', versioninfo=('', '', ''), machine='arm64'):
 
     """ Get macOS version information and return it as tuple (release,
         versioninfo, machine) with versioninfo being a tuple (version,
