diff --git a/configure b/configure
index cbc1ebf..d87ab8e 100755
--- a/configure
+++ b/configure
@@ -19419,7 +19419,7 @@ fi
 # Note 1: test -ef might not exist, but it's more reliable than `pwd`.
 # Note 2: /bin/pwd might be better than shell's built-in at getting
 #         a symlink-free name.
-if ( test "$srcdir" -ef . ) >/dev/null 2>&1 || test "`cd $srcdir && /bin/pwd`" = "`/bin/pwd`"; then
+if ( test "$srcdir" -ef . ) >/dev/null 2>&1 || test "`cd $srcdir && readlink -f /proc/self/cwd`" = "`readlink -f /proc/self/cwd`"; then
   vpath_build=no
 else
   vpath_build=yes
