diff --git a/common.mk b/common.mk
--- a/common.mk
+++ b/common.mk
@@ -1523,17 +1523,6 @@ extract-gems: $(HAVE_BASERUBY:yes=update-gems) $(HAVE_BASERUBY:yes=outdate-bund

 update-gems$(sequential): PHONY
 	$(ECHO) Downloading bundled gem files...
-	$(Q) $(BASERUBY) -C "$(srcdir)" \
-	    -I./tool -rdownloader $(split_option) -answ \
-	    -e 'gem, ver = *$$F' \
-	    -e 'next if !ver' \
-	    -e 'old = Dir.glob("gems/#{gem}-*.gem")' \
-	    -e 'gem = "#{gem}-#{ver}.gem"' \
-	    -e 'Downloader::RubyGems.download(gem, "gems", nil) and' \
-	    -e '(old.delete("gems/#{gem}"); !old.empty?) and' \
-	    -e 'File.unlink(*old) and' \
-	    -e 'FileUtils.rm_rf(old.map{'"|n|"'n.chomp(".gem")})' \
-	    gems/bundled_gems

 extract-gems$(sequential): PHONY
 	$(ECHO) Extracting bundled gem files...
