diff -ur a/common.mk b/common.mk
--- a/common.mk	2023-06-20 12:55:39.909678948 -0500
+++ b/common.mk	2023-06-20 12:56:06.221960421 -0500
@@ -1379,17 +1379,6 @@

 update-gems$(gnumake:yes=-sequential): PHONY
 	$(ECHO) Downloading bundled gem files...
-	$(Q) $(BASERUBY) -C "$(srcdir)" \
-	    -I./tool -rdownloader -answ \
-	    -e 'gem, ver = *$$F' \
-	    -e 'next if !ver or /^#/=~gem' \
-	    -e 'old = Dir.glob("gems/#{gem}-*.gem")' \
-	    -e 'gem = "#{gem}-#{ver}.gem"' \
-	    -e 'Downloader::RubyGems.download(gem, "gems", nil) and' \
-	    -e '(old.delete("gems/#{gem}"); !old.empty?) and' \
-	    -e 'File.unlink(*old) and' \
-	    -e 'FileUtils.rm_rf(old.map{'"|n|"'n.chomp(".gem")})' \
-	    gems/bundled_gems

 extract-gems$(gnumake:yes=-sequential): PHONY
 	$(ECHO) Extracting bundled gem files...
