diff -Naur perl-5.38.2/hints/darwin.sh perl-5.38.2_patched/hints/darwin.sh
--- perl-5.38.2/hints/darwin.sh	2023-11-28 06:57:28
+++ perl-5.38.2_patched/hints/darwin.sh	2023-12-10 14:18:41
@@ -325,7 +325,7 @@
     # sw_vers output                 what we want
     # "ProductVersion:    10.10.5"   "10.10"
     # "ProductVersion:    10.11"     "10.11"
-        prodvers=`sw_vers|awk '/^ProductVersion:/{print $2}'|awk -F. '{print $1"."$2}'`
+        prodvers="${MACOSX_DEPLOYMENT_TARGET:-15.1}"
     case "$prodvers" in
     [1-9][0-9].*)
       add_macosx_version_min ccflags $prodvers
